peekingduck/pipeline/nodes/model/fairmotv1/fairmot_files/utils.py

```diff
@@ -0,0 +1,263 @@
+    `indices` at dim=1.
+
+    Args:
+        feat (torch.Tensor): The tensor containing the features of interest.
+        indices (torch.Tensor): The indices of elements to gather.
+
+    Returns:
+        (torch.Tensor): The transposed and gathered features.
+    """
+    feat = feat.permute(0, 2, 3, 1).contiguous()
+    feat = feat.view(feat.size(0), -1, feat.size(3))
+    feat = gather_feat(feat, indices)
+    return feat
+
+
+def _affine_transform(point: np.ndarray, trans_matrix: np.ndarray) -> np.ndarray:
+    """Applies affine transformation to the specified coordinates.
+
+    Args:
+        point (np.ndarray): The coordinates to transform.
+        trans_matrix (np.ndarray): A 2x3 affine transformation matrix.
+
+    Returns:
+        (np.ndarray): The transformed coordinates.
+    """
+    new_pt = np.array([point[0], point[1], 1.0], dtype=np.float32).T
+    new_pt = trans_matrix @ new_pt
+    return new_pt[:2]
+
+
+def _get_3rd_point(point_1: np.ndarray, point_2: np.ndarray) -> np.ndarray:
+    """Computes the third point required for computing affine transformation
+    matrix.
+
+    Args:
+        point_1 (np.ndarray): The first point.
+        point_2 (np.ndarray): The second point.
+
+    Returns:
+        (np.ndarray): The third point.
+    """
+    direct = point_1 - point_2
+    return point_2 + np.array([-direct[1], direct[0]], dtype=np.float32)
+
+
+def _get_affine_transform(  # pylint:disable=too-many-arguments
+    center: np.ndarray,
+    scale: float,
+    rot: float,
+    output_size: np.ndarray,
+    shift: np.ndarray = np.array([0, 0], dtype=np.float32),
+    inv: bool = False,
+) -> np.ndarray:
+    """Computes the affine transformation matrix described by the given
+    arguments.
+
+    Args:
+        center (np.ndarray): Coordinates of the center.
+        scale (float): Scale factor.
+        rot (float): Rotation amount (in degrees).
+        output_size (np.ndarray): Height and width of the output.
+        shift (np.ndarray): Translation.
+        inv (bool): Flag to determine if we should perform the inverse of the
+            transformation.
+
+    Returns:
+        (np.ndarray): The affine transformation matrix.
+    """
+    src_w = scale
+    dst_w = output_size[0]
+
+    src_dir = _get_dir([0, src_w * -0.5], np.pi * rot / 180)
+    dst_dir = np.array([0, dst_w * -0.5], np.float32)
+
+    src = np.zeros((3, 2), dtype=np.float32)
+    dst = np.zeros((3, 2), dtype=np.float32)
+    src[0, :] = center + scale * shift
+    src[1, :] = center + src_dir + scale * shift
+    dst[0, :] = output_size * 0.5
+    dst[1, :] = output_size * 0.5 + dst_dir
+
+    src[2, :] = _get_3rd_point(src[0, :], src[1, :])
+    dst[2, :] = _get_3rd_point(dst[0, :], dst[1, :])
+
+    if inv:
+        matrix = cv2.getAffineTransform(dst, src)
+    else:  # pragma: no cover
+        # Only inv=True is used by FairMOT
+        matrix = cv2.getAffineTransform(src, dst)
```

Comment:
If always `inv = True`, consider making `inv: bool = True` in line 201.

Reply:
This function is traced back to repos such as [CenterNet](https://github.com/xingyizhou/CenterNet/blob/master/src/lib/utils/image.py#L27) and [human-pose-estimation](https://github.com/microsoft/human-pose-estimation.pytorch/blob/master/lib/utils/transforms.py#L57) which has `inv=False` as the default.

Would it be advisable to keep the default param values/signature consistent with those functions so users familiar with them will not be thrown off guard?

Reply:
Ok, since it's due to compatibility reasons.